src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -305,6 +305,15 @@ private String postTestWebResource() throws Exception {
         return mvcResult.getResponse().getHeader("Location");
     }
 
+    @Test
+    public void postWebResourceNullURL() throws Exception {
```

Comment:
Could you please add another test that fails for a non-empty, but still invalid URL?